internal/metrics/metrics.go

```diff
@@ -77,6 +94,84 @@ func (m Metrics) CrawledPagesCounter() prometheus.Counter {
 	return m.crawledPagesCounter
 }
 
+func fetchMirrorAvailabilityMetric(backend string, url string) (httpStatus int, err error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return
+	}
+	req.Header.Set("Backend-Override", backend)
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return
+	}
```

Comment:
We don't close the response body here, which will cause a memory leak. Please add a defer. `defer resp.Body.Close()`

Reply:
ok, I'll add the defer